tests/sentry/hybrid_cloud/test_organizationmapping.py

```diff
@@ -82,11 +83,74 @@ def test_update_customer_id(self):
         api_org_mapping = organization_mapping_service.create(**fields)
         assert api_org_mapping.customer_id is None
 
-        assert (
-            organization_mapping_service.update_customer_id(
-                organization_id=self.organization.id, customer_id="test"
+        organization_mapping_service.update(
+            ApiOrganizationMappingUpdate(
+                organization_id=self.organization.id,
+                customer_id="test",
             )
-            == 1
         )
         org_mapping = OrganizationMapping.objects.get(organization_id=self.organization.id)
         assert org_mapping.customer_id == "test"
+
+        organization_mapping_service.update(
+            ApiOrganizationMappingUpdate(organization_id=self.organization.id, name="new name!")
+        )
+        org_mapping = OrganizationMapping.objects.get(organization_id=self.organization.id)
+        assert org_mapping.customer_id == "test"
+        assert org_mapping.name == "new name!"
+
+        organization_mapping_service.update(
+            ApiOrganizationMappingUpdate(organization_id=self.organization.id)
+        )
+        # Does not overwrite with empty value.
+        assert org_mapping.name == "new name!"
+
+    def test_create_org_with_name(self):
+        # snowflake id generation isn't great.  We need this to prevent flaky tests.
+        time.sleep(1)
+
+        # This is much how a hybrid cloud org creation + slug reservation would work.
+        # Notice that org objects are not created before, rather after org slug reservation.
+        org_id = Organization.reserve_snowflake_id()
```

Comment:
It's pretty trivial to generate and reserve unique ids that are guaranteed not to collide, actually.  Our current snowflake id implementation is just, wrong, and can be fixed.  I intend to do so.

We need the id to reserve the slug before org creation, correct?  Otherwise, we end up creation orgs that are missing slugs, and then the need to reconcile that state?  And it's really weird for the user, too.

Reply:
Yes, ids are needed to reserve the slug. My thinking was that we could get the id from the row after it was created in the uncommitted transaction. The flow would be similar to

```python
with transaction.atomic():
   org.save()
   organization_mapping_service.create(user=user, organization_id=org_id, slug='cool-slug', ...)
```

If the request to reserve the slug were to fail we would rollback the transaction. This would could result in slugs being reserved but not used though due to response timeouts. We have a cleanup/repair task for dangling slugs though.

Pre-reserving slugs will also work though :smile: